narwhal/worker/src/handlers.rs

```diff
@@ -77,6 +80,30 @@ impl<V: TransactionValidator> WorkerToWorker for WorkerReceiverHandler<V> {
 
         Ok(anemo::Response::new(RequestBatchResponse { batch }))
     }
+
+    async fn request_batches(
+        &self,
+        request: anemo::Request<RequestBatchesRequest>,
+    ) -> Result<anemo::Response<RequestBatchesResponse>, anemo::rpc::Status> {
+        let batch_digests = request.into_body().batches;
```

Comment:
think it's worth adding a limit on `batch_digests.len()` in addition to size, to prevent doing excess work in case of a badly behaved peer? (esp because you fetch all of the requested batches regardless of how many are returned). worst-case, this could cause us to read from the DB 500KB*number of requested batches, right?

Reply:
Agreed, added a relatively high limit of 100 batches for now. Let me know if you think that limit is too high, need to get more data to make a more informed decision. But should protect us from any malicious peers.

Reply:
Turns out 100 was too low, updated it to 1_000